Document manual Xcode project migration to support Xcode 11.4 (#3763)
flutter/website: src/docs/development/ios-project-migration.md

---
title: Supporting Xcode 11.4
description: How to migrate existing Flutter iOS projects to Xcode 11.4.
---

{{site.alert.note}}

>   You might be directed to this page if the tool detects that your project

Author: As you noticed, I copied this from the AndroidX page.  It's not from the framework, it's coming from running `flutter build ios` on the command line or an error in an IDE.  What about
> You might be directed to this page if Flutter detects

  doesn't support Xcode 11.4.
{{site.alert.end}}

> To developer Flutter apps for iOS, you need a Mac with Xcode.

Author: Will update https://github.com/flutter/website/blob/master/src/docs/get-started/install/_ios-setup.md#install-xcode as well.

Xcode 11.4 changes the way frameworks are linked and embedded, and
you may [experience issues switching between iOS devices and simulators][].
Flutter v1.15.3 and later will automatically migrate your Xcode project.

If you need to manually upgrade:
1. From the Flutter app directory, open `ios/Runner.xcworkspace` in Xcode.
1. In the Navigator pane, locate the Flutter group and remove App.framework
and Flutter.framework.
{% include app-figure.md image="development/ios-project-migration/navigator.png" alt="Remove Frameworks in Xcode Navigator" %}

1. In the Runner target build settings > Build Phases > Link Binary With Libraries

> confirm App.framework and Flutter.framework are no longer present. Also confirm

Reviewer: ```
=> confirm that `App.framework` and `Flutter.framework` are...
```

> in Build Phases > Embed Frameworks.

Reviewer: ```
=> Also confirm that these files no longer appear in **Build Phases > Embed Frameworks**.
```

{% include app-figure.md image="development/ios-project-migration/framework-build-phase.png" alt="Confirm Frameworks Removed from Build Phases" %}

> 1. The Runner target build settings > Build Phases > Thin Binary script becomes:

Reviewer: ```
=> **Build Phases > Thin Binary** script
```

Also, this sentence is not clear to me. Does this setting change automatically, or do they change it? How about:

```
=> Change the Runner target build settings > **Build Phases > Thin Binary** script as follows:
```

```sh
/bin/sh "$FLUTTER_ROOT/packages/flutter_tools/bin/xcode_backend.sh" embed
/bin/sh "$FLUTTER_ROOT/packages/flutter_tools/bin/xcode_backend.sh" thin
```
{% include app-figure.md image="development/ios-project-migration/script-phase.png" alt="Update Thin Binary Script Build Phase" %}

> 1. In the Runner target build settings > Build Settings > Other Linker Arguments (`OTHER_LDFLAGS`)

Reviewer: ```
=> **Build Settings > Other Linker Arguments**
```

(Basically, the Google developer docs rule is that UI text is bolded. Files are generally code fonted.

---
title: Xcode 11.4 Support
description: How to migrate existing Flutter iOS projects to Xcode 11.4.
---

{{site.alert.note}}
  You might be directed to this page if Flutter detects that your project

>   doesn't support Xcode 11.4.

Reviewer 2: detects that you have Xcode 11.4 installed AND your project doesn't support Xcode 11.4?

Author: Yes, the tool gives the error message when the migration fails AND Xcode 11.4 or greater is installed.  We probably don't need to point that out, either they saw the error or they didn't.

{{site.alert.end}}

To develop Flutter apps for iOS, you need a Mac with Xcode installed.
Xcode 11.4 changed the way frameworks are linked and embedded, and
you may [experience issues switching between iOS devices and simulators][].
Flutter v1.15.3 and later will automatically migrate your Xcode project.

If you need to manually upgrade your project, use the following steps:

> <ol markdown="1">

Reviewer 2: FWIW, this is barely documented but you can unbreak the weird markdown ordered list thing with big blocks if you 4 space indent everything i.e. 

```
1. some thing

    `some long code section which would have made a mess`
1. this should be item 2 and should work now
```

Author: Interesting!  I don't know if @sfshaza2 tried that, but the `ol` markdown was how she got  https://github.com/flutter/website/blob/master/src/docs/development/add-to-app/ios/project-setup.md to work.  She asked me to do it here so I'll stick with that for now, but I can follow up and experiment when she's back in the office next week.

> <li markdown="1">From the Flutter app directory, open `ios/Runner.xcworkspace` in Xcode.

Reviewer 2: Does xcworkspace exist if the project doesn't use plugins? I don't remember.

Author: Yes, it's always added as part of the create template.

</li>
<li markdown="1">In the **Navigator** pane, locate the **Flutter** group and remove `App.framework`
and `Flutter.framework`.
{% include app-figure.md image="development/ios-project-migration/navigator.png" alt="Remove Frameworks in Xcode Navigator" %}
</li>

> <li markdown="1">In the Runner target build settings > **Build Phases > Link Binary With Libraries**

Reviewer 3: ```suggestion
<li markdown="1">Navigate to the Runner target build settings  **Build Phases > Link Binary With Libraries**
```

> <li markdown="1">In the Runner target build settings > **Build Phases > Link Binary With Libraries**

Reviewer 3: ```suggestion
<li markdown="1">In the Runner target build settings **Build Phases > Link Binary With Libraries**
```

confirm `App.framework` and `Flutter.framework` are no longer present. Also confirm
in **Build Phases > Embed Frameworks**.
{% include app-figure.md image="development/ios-project-migration/framework-build-phase.png" alt="Confirm Frameworks Removed from Build Phases" %}
</li>

> <li markdown="1">Change the Runner target build settings > **Build Phases > Thin Binary** script as follows:

Reviewer 3: ```suggestion
<li markdown="1">Change the Runner target build settings **Build Phases > Thin Binary** script as follows:
```
